model/src/order.rs

```diff
@@ -202,19 +190,45 @@ impl OrderCreation {
 // Intended to be used by tests that need signed orders.
 impl OrderCreation {}
 
-// See https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2Encoding.sol
 impl OrderCreation {
+    // See https://github.com/gnosis/gp-v2-contracts/blob/main/src/contracts/libraries/GPv2Encoding.sol
     pub const ORDER_TYPE_HASH: [u8; 32] =
         hex!("b2b38b9dcbdeb41f7ad71dea9aed79fb47f7bbc3436576fe994b43d5b16ecdec");
-
     // keccak256("sell")
     const ORDER_KIND_SELL: [u8; 32] =
         hex!("f3b277728b3fee749481eb3e0b3b48980dbbab78658fc419025cb16eee346775");
     // keccak256("buy")
     const ORDER_KIND_BUY: [u8; 32] =
         hex!("6ed88e868af0a1983e3886d5f3e95a2fafbd6c3450bc229e27342283dc429ccc");
+}
+
+impl Default for OrderCancellation {
+    // Custom implementation to make sure the default order is valid
+    fn default() -> Self {
+        let mut result = Self {
+            order_uid: OrderUid::default(),
+            signature: Default::default(),
+        };
+        result.sign_self_with(&DomainSeparator::default(), &SecretKeyRef::new(&ONE_KEY));
+        result
+    }
+}
 
-    fn order_digest(&self) -> [u8; 32] {
+impl OrderCancellation {
+    fn sign_self_with(&mut self, domain_separator: &DomainSeparator, key: &SecretKeyRef) {
```

Comment:
nit: I feel like this logic could now also be implemented in the trait.

Reply:
Very clever! Excellent observation.

Reply:
Although, I am not sure though because the last three lines are actually setting the signature... I would have to implement a set_signature method or something on both... or maybe change things so that this returns a signature to be set outside:

```rust
    fn sign_self_with(&mut self, domain_separator: &DomainSeparator, key: &SecretKeyRef) {
        let message = self
            .signature()
            .signing_digest_message(domain_separator, &self.digest());
        // Unwrap because the only error is for invalid messages which we don't create.
        let signature = Key::sign(key, &message, None).unwrap();
        self.signature.v = signature.v as u8 | 0x80;
        self.signature.r = signature.r;
        self.signature.s = signature.s;
    }
```

Reply:
Ok, I think I have found something nice that works well and reduces redundancy much more.